nservicebus/nhibernate/custom-mappings.md

```diff
@@ -0,0 +1,153 @@
+---
+title: NHibernate Custom mapping with hbm.xml, attributes or fluent api
+summary: Create a custom mapping to change your NHibernate creates the database schema using different techniques.
+tags:
+ - NHibernate
+ - Persistence
+ - Configuration
+ - Mapping
+related:
+ - nservicebus/nhibernate/accessing-data
+ - nservicebus/nhibernate/index
+---
+
+
+Sometimes the default schema that gets generated for your saga does work. For example, you might need to store strings with a different string length, want so use complex types or want to store DateTime values with a high precision or maybe want to tweak the eager or lazy loading to improve performance. You will need to create a custom mapping to achieve this.
+
+Custom mapping options:
+
+* Create a `*.hbm.xml` file for your saga data class
+* Use Fluent NHibernate (popular seperate fluent api)
+* Use NHibernate.Mapping.Attributes
+* Loquacious Configuration (native fluent api)
+
+There are probably even other options but these are the most frequently used.
+
+See http://docs.particular.net/nservicebus/nhibernate/configuration on how you can pass this configuration object to the NServiceBus NHibernate configuration.
```

Comment:
@ramonsmits links need to be relative or fully qualified by starting with a "/" https://github.com/Particular/docs.particular.net#linking  
if you install the markdownpad hack it will perform these checks for you